examples/tfengine/folder_foundation.hcl

```diff
@@ -55,32 +55,35 @@ template "cicd" {
       owner = "GoogleCloudPlatform"
       name  = "example"
     }
-    branch_name    = "master"
-    terraform_root = "terraform"
 
     # Required to create Cloud Scheduler jobs.
     scheduler_region = "us-east1"
 
-    # Prepare and enable default triggers.
-    triggers = {
-      validate = {}
-      plan     = {
-        run_on_schedule = "0 12 * * *" # Run at 12 PM EST everyday
-      }
-      apply    = {
-        run_on_schedule = "0 13 * * *" # Run at 13 PM EST everyday
-      }
-    }
-
     build_viewers = [
       "group:example-cicd-viewers@example.com",
     ]
-
-    managed_dirs = [
-      "devops", // NOTE: CICD service account can only update APIs on the devops project.
-      "audit",
-      "monitor",
-      "folders",
+    terraform_root = "terraform"
+    envs = [
+      {
+        name        = "prod"
+        branch_name = "main"
+        triggers = {
+          validate = {}
+          plan = {
+            run_on_schedule = "0 12 * * *" # Run at 12 PM EST everyday
+          }
+          apply = {
+            run_on_push = false # Do not auto run on push to branch
+          }
+        }
+        managed_dirs = [
```

Comment:
We have to duplicate managed_dirs for all envs?

Reply:
managed dirs is a bit tricky with multi-env especially due to the ones shared between envs (devops, audit), and the paths for different envs are not fixed at this moment. This way it gives the most flexibility.

Reply:
Can we have a global order and then just cd through and check if they exist (and deploy if they do)?

Reply:
Anyways, that doesn't need to be solved now. I will solve it as part of the app recipe.